src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -409,17 +420,59 @@ def _process_heatmap_tools(heatmap_tools: set[str],
         _create_import_files(root_dir, rows_output, heatmap_rows_list)
         _create_import_files(root_dir, cell_details_output, heatmap_cell_details_list)
 
-        if bad_data_ids:
-            result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool=tool)
-            with open(os.path.join(result_dir, BAD_DATA_IDS), "w") as outfile:
-                json.dump(bad_data_ids, outfile)
+
+def _process_fatal_error_tools(check_fatal_error_tools: set[str],
+                               root_dir: str,
+                               kbase_collection: str,
+                               load_ver: str):
+    # process fatal error files from a set of check tools and write it to merged file
+
+    if not check_fatal_error_tools:
+        return set()
+    
+    fatal_dict = dict()
+    for tool in check_fatal_error_tools:
+        result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool=tool)
+        batch_dirs = _get_batch_dirs(result_dir)
+        if len(batch_dirs) == 1 and batch_dirs[0].startswith(loader_common_names.COMPUTE_OUTPUT_NO_BATCH_PREFIX):
+            batch_dirs = [os.path.join(batch_dirs[0], d) for d in os.listdir(batch_dirs[0])
+                          if os.path.isdir(os.path.join(result_dir, batch_dirs[0], d))]
+        for batch_dir in batch_dirs:
+            data_dir = os.path.join(result_dir, batch_dir)
+            fatal_error_file = os.path.join(data_dir, loader_common_names.FATAL_ERROR_FILE)
+            if not os.path.exists(fatal_error_file):
+                continue
+            try:
+                with open(fatal_error_file, "r") as json_file:
+                    fatal_errors = json.load(json_file)
+            except Exception as e:
+                raise ValueError(f"{fatal_error_file} exists, but unable to retrive") from e
+            
+            for kbase_id in fatal_errors:
+                fatal_dict_info = create_global_fatal_dict_doc(
+                    tool, 
+                    fatal_errors[kbase_id][loader_common_names.FATAL_ERROR], 
+                    fatal_errors[kbase_id][loader_common_names.FATAL_STACKTRACE])
+                if fatal_dict.get(kbase_id):
+                    fatal_dict[kbase_id][loader_common_names.FATAL_ERRORS].append(
+                        fatal_dict_info)
+                else:
+                    fatal_dict[kbase_id] = {loader_common_names.FATAL_FILE: fatal_errors[kbase_id][loader_common_names.FATAL_FILE],
+                                            loader_common_names.FATAL_ERRORS: [fatal_dict_info]}
+    
+    docs = [{key:value} for key, value in fatal_dict.items()]
+    output = f"{kbase_collection}_{load_ver}_{FATAL_ERROR_FILE_SUFFIX}"
+    _create_import_files(root_dir, output, docs)
```

Comment:
This isn't going to be able to be loaded into arango as is and is going to be hard to read. I'd recommend going back to the dict format keyed by the ID for now.

Reply:
Reverted back!

Reply:
:+1: